Cython/Compiler/ExprNodes.py

```diff
@@ -10208,7 +10208,12 @@ def generate_codeobj(self, code):
         func_name_result = code.get_py_string_const(
             func.name, identifier=True, is_str=False, unicode_value=func.name)
         # FIXME: better way to get the module file path at module init time? Encoding to use?
-        file_path = StringEncoding.bytes_literal(func.pos[0].get_filenametable_entry().encode('utf8'), 'utf8')
+        file_path = func.pos[0].get_filenametable_entry()
+        if os.path.isabs(file_path):
+            file_path = func.pos[0].get_description()
+        # Always use / as separator
+        file_path = file_path.replace('\\', '/')
```

Comment:
The [`.as_posix()`](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.as_posix) method in `pathlib` seems a better choice here because this should only be done on Windows. A Unix path can contain `\` as character:
```bash
$ touch abcdefg\\xyz
$ ls abcdefg\\xyz
'abcdefg\xyz'
```
OTOH, Windows directory names can actually contain `/`, and `.as_posix()` doesn't try to escape that either.

It seems slightly fragile to "normalise" path names like this, but I guess users would try to avoid them – at least after running into them once.